src/tasks/minions/farmingActivity.ts

```diff
@@ -167,8 +168,17 @@ export const farmingTask: MinionTask = {
 			}
 			checkHealthXp = alivePlants * plantToHarvest.checkXp;
 
+			const cleanHerb =
+				plantToHarvest.herbXp !== undefined &&
+				user.bitfield.includes(BitField.CleanHerbsFarming) &&
+				user.skillLevel(SkillsEnum.Herblore) >= plantToHarvest.herbLvl!;
+
 			if (plantToHarvest.givesCrops) {
-				if (!plantToHarvest.outputCrop) return;
+				let cropToHarvest = plantToHarvest.outputCrop;
+				if (cleanHerb) {
+					cropToHarvest = plantToHarvest.cleanHerbCrop;
+				}
+				if (!cropToHarvest) return;
```

Comment:
In what scenario, or why, are we returning here? Is it an error-case? if so, it should throw an error, or have a test.